src/panels/config/entities/entity-registry-settings-editor.ts

```diff
@@ -609,32 +616,48 @@ export class EntityRegistrySettingsEditor extends LitElement {
             @selected=${this._switchAsChanged}
             @closed=${stopPropagation}
           >
-            <ha-list-item
-              value="switch"
-              .selected=${!this._deviceClass || this._deviceClass === "switch"}
-            >
-              ${this.hass.localize(
-                "ui.dialogs.entity_registry.editor.device_classes.switch.switch"
-              )}
-            </ha-list-item>
-            <ha-list-item
-              value="outlet"
-              .selected=${this._deviceClass === "outlet"}
-            >
-              ${this.hass.localize(
-                "ui.dialogs.entity_registry.editor.device_classes.switch.outlet"
-              )}
-            </ha-list-item>
+            ${domain === "switch"
+              ? html`<ha-list-item
+                    value="switch"
+                    .selected=${!this._deviceClass ||
+                    this._deviceClass === "switch"}
+                  >
+                    ${this.hass.localize(
+                      "ui.dialogs.entity_registry.editor.device_classes.switch.switch"
+                    )}
+                  </ha-list-item>
+                  <ha-list-item
+                    value="outlet"
+                    .selected=${this._deviceClass === "outlet"}
+                  >
+                    ${this.hass.localize(
+                      "ui.dialogs.entity_registry.editor.device_classes.switch.outlet"
+                    )}
+                  </ha-list-item>`
+              : html`<ha-list-item
+                    value="switch"
+                    .selected=${this._switchAs === "switch"}
+                  >
+                    ${domainToName(this.hass.localize, "switch")}
+                  </ha-list-item>
+                  <ha-list-item
+                    value="switch"
```

Comment:
```suggestion
                    .value=${domain}
```
Shouldn't be domain here ?